crates/evm/src/trace/identifier/local.rs

```diff
@@ -1,56 +1,40 @@
 use super::{AddressIdentity, TraceIdentifier};
-use ethers::{
-    abi::{Abi, Address, Event},
-    prelude::ArtifactId,
-};
+use ethers::abi::{Address, Event};
 use foundry_common::contracts::{diff_score, ContractsByArtifact};
-use itertools::Itertools;
 use ordered_float::OrderedFloat;
-use std::{borrow::Cow, collections::BTreeMap};
+use std::borrow::Cow;
 
 /// A trace identifier that tries to identify addresses using local contracts.
-pub struct LocalTraceIdentifier {
-    local_contracts: BTreeMap<Vec<u8>, (ArtifactId, Abi)>,
+pub struct LocalTraceIdentifier<'a> {
+    known_contracts: &'a ContractsByArtifact,
 }
 
-impl LocalTraceIdentifier {
-    pub fn new(known_contracts: &ContractsByArtifact) -> Self {
-        Self {
-            local_contracts: known_contracts
-                .iter()
-                .map(|(id, (abi, runtime_code))| (runtime_code.clone(), (id.clone(), abi.clone())))
-                .collect(),
-        }
+impl<'a> LocalTraceIdentifier<'a> {
+    pub fn new(known_contracts: &'a ContractsByArtifact) -> Self {
+        Self { known_contracts }
```

Comment:
Remove useless re-allocations, we only need to be able to iterate over the 3 values (id, abi, code)